### DISCUSSION TITLE
'Set-PnPDefaultColumnValues' not working on sharepoint online library

### DISCUSSION THREAD

**Discussion by sav169** [pnp/powershell#1883] [Q&A]
Set-PnPDefaultColumnValues not working.
- I have a sharepoint online subsite 
- I have a library called 'mappings' 
- I have the following folders in the library- A B C D 
- I have a managed metedata column for the library called 'map' 
- I want to set the default value for all the above folders for the column 'Map' using the powershell command. 
- I am running the below command to set the default value but it does not set the default value. 
- There is no error. The command completes but it does not set the default value on the folder. 
- The command I run is below
 Set-PnPDefaultColumnValues -List ["mappings"] -Field [map] -Folder "D" -Value "Information Technology|mappings|D"
- This works if I do it through UI via the library settings - Column default value settings but since I have to map a lot of folders in multiple libraries, I want to use Powershell

**Comment by denicomp** [comment]
I just followed your exact steps and it worked OK for me. Were you able to get this working?

I have found that if you mess up with Set-PnpDefaultColumnValues, it can irreparably screw up your library, to the point where the only way to fix it is to delete it and re-create it.

For example, I thought I might be able to use a Lookup column and set a default value on it for a folder. The command doesn't complain, but then it doesn't work and you can't remove it. I think I had a similar issue where I misspelled the name of the term group - it didn't give me an error, but I couldn't undo it.

**Reply by sav169** [reply]
Thanks for having a look. For some reason it doesnt work for me. tried on multiple libraries and sites.
